### PR TITLE
Add support of events

### PR DESCRIPTION
**Pull request by marcosmc**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
The dynamic-simulation does not support events file.

**What is the new behavior (if this is a feature change)?**
The events file can be passed to the command, and loaded. Only groovy is supported yet.

### CODE REVIEW

File: dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java [hunk 1]
```diff
@@ -49,6 +52,7 @@
 
     private static final String CASE_FILE = "case-file";
     private static final String DYNAMIC_MODEL_FILE = "dynamic-model-file";
```
Comment (mathbagu) [inline dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java]: Rename this constant DYNAMIC_MODELS_FILE

File: dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java [hunk 2]
```diff
@@ -49,6 +52,7 @@
...
+    private static final String DYNAMIC_EVENT_MODEL_FILE = "events-file";
```
Comment (mathbagu) [inline dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java:55]: Rename this constant EVENTS_MODEL_FILE, and the parameter `events-model-file` to be consistent

File: dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java [hunk 3]
```diff
@@ -49,6 +52,7 @@
 
     private static final String CASE_FILE = "case-file";
     private static final String DYNAMIC_MODEL_FILE = "dynamic-models-file";
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java:54]: @marcosmc could you also rename the variable DYNAMIC_MODEL_FILE into DYNAMIC_MODELS_FILE? I forgot to do it in my PR. Thanks.

File: dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java [hunk 4]
```diff
@@ -149,7 +164,7 @@ public void run(CommandLine line, ToolRunningContext context) throws Exception {
             JsonDynamicSimulationParameters.update(params, parametersFile);
         }
 
-        DynamicSimulationResult result = runner.run(network, dydSupplier, curvesSupplier, VariantManagerConstants.INITIAL_VARIANT_ID, context.getShortTimeExecutionComputationManager(), params);
+        DynamicSimulationResult result = runner.run(network, dydSupplier, eventSupplier, curvesSupplier, VariantManagerConstants.INITIAL_VARIANT_ID, context.getShortTimeExecutionComputationManager(), params);
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java:167]: @marcosmc could you change also dydSupplier into dynamicModelsSupplier please?

File: dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicSimulation.java [hunk 5]
```diff
@@ -32,24 +32,36 @@ public Runner(DynamicSimulationProvider provider) {
...
+
+        public CompletableFuture<DynamicSimulationResult> runAsync(Network network, DynamicModelsSupplier dynamicModelsSupplier, DynamicEventModelsSupplier dynamicEventModelsSupplier, CurvesSupplier curvesSupplier, DynamicSimulationParameters parameters) {
+            return runAsync(network, dynamicModelsSupplier, dynamicEventModelsSupplier, curvesSupplier, network.getVariantManager().getWorkingVariantId(), parameters);
+        }
+
+        public CompletableFuture<DynamicSimulationResult> runAsync(Network network, DynamicModelsSupplier dynamicModelsSupplier, DynamicEventModelsSupplier dynamicEventModelsSupplier, DynamicSimulationParameters parameters) {
+            return runAsync(network, dynamicModelsSupplier, dynamicEventModelsSupplier, CurvesSupplier.empty(), parameters);
         }
 
         public CompletableFuture<DynamicSimulationResult> runAsync(Network network, DynamicModelsSupplier dynamicModelsSupplier, CurvesSupplier curvesSupplier, DynamicSimulationParameters parameters) {
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicSimulation.java:53]: Here if we follow the combinations' specification we sent, the parameters should be removed.

File: dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java (hunk 3 above)
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java]: Done

File: dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java (hunk 4 above)
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java]: Done

File: dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicEventModel.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.powsybl.dynamicsimulation;
+
+/**
+ * @author Marcos de Miguel <demiguelm at aia.es>
+ */
+public interface DynamicEventModel {
```
[the remark this answers is not in the document]
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicEventModel.java]: Done

File: dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicEventModelsSupplier.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+ */
+
+package com.powsybl.dynamicsimulation;
+
+import java.util.Collections;
+
+/**
+ * @author Marcos de Miguel <demiguelm at aia.es>
+ */
+public interface DynamicEventModelsSupplier extends SimulatorInputSupplier<DynamicEventModel> {
```
[the remark this answers is not in the document]
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicEventModelsSupplier.java]: Done

File: dynamic-simulation/dynamic-simulation-dsl/src/main/java/com/powsybl/dynamicsimulation/groovy/DynamicEventModelGroovyExtension.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+ */
+
+package com.powsybl.dynamicsimulation.groovy;
+
+import com.powsybl.dynamicsimulation.DynamicEventModel;
+
+/**
+ * @author Marcos de Miguel <demiguelm at aia.es>
+ */
+public interface DynamicEventModelGroovyExtension extends GroovyExtension<DynamicEventModel> {
```
[the remark this answers is not in the document]
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-dsl/src/main/java/com/powsybl/dynamicsimulation/groovy/DynamicEventModelGroovyExtension.java]: Done

File: dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java (hunk 1 above)
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java]: Done

File: dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java (hunk 2 above)
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java]: Done

File: dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicSimulation.java [hunk 9]
```diff
@@ -58,13 +70,21 @@ public Runner(DynamicSimulationProvider provider) {
             return runAsync(network, dynamicModelsSupplier, DynamicSimulationParameters.load());
         }
 
-        public DynamicSimulationResult run(Network network, DynamicModelsSupplier dynamicModelsSupplier, CurvesSupplier curvesSupplier, String workingVariantId, ComputationManager computationManager,
+        public DynamicSimulationResult run(Network network, DynamicModelsSupplier dynamicModelsSupplier, EventModelsSupplier dynamicEventModelsSupplier, CurvesSupplier curvesSupplier, String workingVariantId, ComputationManager computationManager,
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicSimulation.java:73]: Please rename also dynamicEventModelsSupplier into eventModelsSupplier everywhere.

File: dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicSimulationProvider.java [hunk 10]
```diff
@@ -18,6 +18,7 @@
  */
 public interface DynamicSimulationProvider extends Versionable, PlatformConfigNamedProvider {
 
-    CompletableFuture<DynamicSimulationResult> run(Network network, DynamicModelsSupplier dynamicModelsSupplier, CurvesSupplier curvesSupplier, String workingVariantId, ComputationManager computationManager, DynamicSimulationParameters parameters);
+    CompletableFuture<DynamicSimulationResult> run(Network network, DynamicModelsSupplier dynamicModelsSupplier, EventModelsSupplier dynamicEventModelsSupplier,
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicSimulationProvider.java:21]: Please rename also dynamicEventModelsSupplier into eventModelsSupplier here.

File: dynamic-simulation/dynamic-simulation-api/src/test/java/com/powsybl/dynamicsimulation/AnotherDynamicSimulationProviderMock.java [hunk 11]
```diff
@@ -27,8 +27,8 @@ public String getVersion() {
     }
 
     @Override
-    public CompletableFuture<DynamicSimulationResult> run(Network network, DynamicModelsSupplier dynamicModelsSupplier, CurvesSupplier curvesSupplier,
-        String workingVariantId, ComputationManager computationManager, DynamicSimulationParameters parameters) {
+    public CompletableFuture<DynamicSimulationResult> run(Network network, DynamicModelsSupplier dynamicModelsSupplier, EventModelsSupplier dynamicEventModelsSupplier,
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-api/src/test/java/com/powsybl/dynamicsimulation/AnotherDynamicSimulationProviderMock.java:30]: Please rename also dynamicEventModelsSupplier into eventModelsSupplier here.

File: dynamic-simulation/dynamic-simulation-api/src/test/java/com/powsybl/dynamicsimulation/DynamicSimulationProviderMock.java [hunk 12]
```diff
@@ -29,8 +29,8 @@ public String getVersion() {
     }
 
     @Override
-    public CompletableFuture<DynamicSimulationResult> run(Network network, DynamicModelsSupplier dynamicModelsSupplier, CurvesSupplier curvesSupplier, String workingVariantId,
-                                                          ComputationManager computationManager, DynamicSimulationParameters parameters) {
+    public CompletableFuture<DynamicSimulationResult> run(Network network, DynamicModelsSupplier dynamicModelsSupplier, EventModelsSupplier dynamicEventModelsSupplier, CurvesSupplier curvesSupplier,
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-api/src/test/java/com/powsybl/dynamicsimulation/DynamicSimulationProviderMock.java:32]: Please rename also dynamicEventModelsSupplier into eventModelsSupplier here.

File: dynamic-simulation/dynamic-simulation-dsl/src/test/groovy/com/powsybl/dynamicsimulation/groovy/DummyDynamicEventModelGroovyExtension.groovy [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+import com.powsybl.dsl.DslException
+import com.powsybl.dynamicsimulation.EventModel
+
+/**
+ * @author Marcos de Miguel <demiguelm at aia.es>
+ */
+@AutoService(EventModelGroovyExtension.class)
+class DummyDynamicEventModelGroovyExtension implements EventModelGroovyExtension {
+
+    static class DummyDynamicEventModelSpec {
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-dsl/src/test/groovy/com/powsybl/dynamicsimulation/groovy/DummyDynamicEventModelGroovyExtension.groovy:21]: Please rename also DummyDynamicEventModel* into DummyEventModel* (even in the file name).

File: dynamic-simulation/dynamic-simulation-dsl/src/test/groovy/com/powsybl/dynamicsimulation/groovy/DummyDynamicEventModelGroovyExtension.groovy [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+        void id(String id) {
+            this.id = id
+        }
+    }
+
+    void load(Binding binding, Consumer<EventModel> consumer) {
+        binding.dummyDynamicEventModel = { Closure<Void> closure ->
+            def cloned = closure.clone()
+
+            DummyDynamicEventModelSpec dynamicEventModelSpec = new DummyDynamicEventModelSpec()
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-dsl/src/test/groovy/com/powsybl/dynamicsimulation/groovy/DummyDynamicEventModelGroovyExtension.groovy:33]: Same here, also for dynamicEventModelSpec -> eventModelSpec

File: dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyDynamicEventModelSupplierTest.java [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+import com.google.common.jimfs.Jimfs;
+import com.powsybl.dynamicsimulation.EventModel;
+import com.powsybl.dynamicsimulation.EventModelsSupplier;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.test.EurostagTutorialExample1Factory;
+
+/**
+ * @author Marcos de Miguel <demiguelm at aia.es>
+ */
+public class GroovyDynamicEventModelSupplierTest {
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyDynamicEventModelSupplierTest.java:32]: Rename also this class -> GroovyEventModelSupplierTest

File: dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyDynamicEventModelSupplierTest.java [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+
+    private FileSystem fileSystem;
+
+    @Before
+    public void setup() throws IOException {
+        fileSystem = Jimfs.newFileSystem(Configuration.unix());
+
+        Files.copy(getClass().getResourceAsStream("/dynamicEventModels.groovy"), fileSystem.getPath("/dynamicEventModels.groovy"));
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyDynamicEventModelSupplierTest.java:40]: Rename also the file eventModels.groovy

File: dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyDynamicEventModelSupplierTest.java [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+        assertEquals(1, extensions.size());
+        assertTrue(extensions.get(0) instanceof EventModelGroovyExtension);
+
+        EventModelsSupplier supplier = new GroovyDynamicEventModelsSupplier(fileSystem.getPath("/dynamicEventModels.groovy"), extensions);
+
+        List<EventModel> dynamicEventModels = supplier.get(network);
+        assertEquals(2, dynamicEventModels.size());
+
+        assertTrue(dynamicEventModels.get(0) instanceof DummyEventModel);
+        DummyEventModel dynamicModel1 = (DummyEventModel) dynamicEventModels.get(0);
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyDynamicEventModelSupplierTest.java:62]: Here also, the naming should be changed. You'll have
DummyEventModel eventModel1 = (DummyEventModel) eventModels.get(0)

File: dynamic-simulation/dynamic-simulation-dsl/src/test/resources/dynamicEventModels.groovy [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+ */
+
+import com.powsybl.iidm.network.Line
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+Logger logger = LoggerFactory.getLogger("com.powsybl.dynamicsimulation.groovy.GroovyDynamicEventModelSupplier")
+
+for (Line line : network.lines) {
+    dummyDynamicEventModel {
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-dsl/src/test/resources/dynamicEventModels.groovy:15]: Here it'll be dummyEventModel

File: dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicSimulation.java (hunk 9 above)
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicSimulation.java]: Done

File: dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicSimulationProvider.java (hunk 10 above)
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicSimulationProvider.java]: Done

File: dynamic-simulation/dynamic-simulation-api/src/test/java/com/powsybl/dynamicsimulation/AnotherDynamicSimulationProviderMock.java (hunk 11 above)
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-api/src/test/java/com/powsybl/dynamicsimulation/AnotherDynamicSimulationProviderMock.java]: Done

File: dynamic-simulation/dynamic-simulation-api/src/test/java/com/powsybl/dynamicsimulation/DynamicSimulationProviderMock.java (hunk 12 above)
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-api/src/test/java/com/powsybl/dynamicsimulation/DynamicSimulationProviderMock.java]: Done

File: dynamic-simulation/dynamic-simulation-dsl/src/test/groovy/com/powsybl/dynamicsimulation/groovy/DummyDynamicEventModelGroovyExtension.groovy (hunk 13 above)
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-dsl/src/test/groovy/com/powsybl/dynamicsimulation/groovy/DummyDynamicEventModelGroovyExtension.groovy]: Done

File: dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyDynamicEventModelSupplierTest.java (hunk 15 above)
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyDynamicEventModelSupplierTest.java]: Done

File: dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyDynamicEventModelSupplierTest.java (hunk 16 above)
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyDynamicEventModelSupplierTest.java]: Done

File: dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyDynamicEventModelSupplierTest.java (hunk 17 above)
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyDynamicEventModelSupplierTest.java]: Done

File: dynamic-simulation/dynamic-simulation-dsl/src/test/resources/dynamicEventModels.groovy (hunk 18 above)
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-dsl/src/test/resources/dynamicEventModels.groovy]: Done

File: dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyEventModelSupplierTest.java [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+        assertEquals(1, extensions.size());
+        assertTrue(extensions.get(0) instanceof EventModelGroovyExtension);
+
+        EventModelsSupplier supplier = new GroovyEventModelsSupplier(fileSystem.getPath("/eventModels.groovy"), extensions);
+
+        List<EventModel> eventModels = supplier.get(network);
+        assertEquals(2, eventModels.size());
+
+        assertTrue(eventModels.get(0) instanceof DummyEventModel);
+        DummyEventModel dynamicModel1 = (DummyEventModel) eventModels.get(0);
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyEventModelSupplierTest.java:62]: Here dynamicModel1 and dynamicModel2 should be eventModel1 and eventModel2

Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-dsl/src/test/java/com/powsybl/dynamicsimulation/groovy/GroovyEventModelSupplierTest.java]: Done

File: dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java [hunk 20]
```diff
@@ -134,8 +143,14 @@ public void run(CommandLine line, ToolRunningContext context) throws Exception {
 
         DynamicSimulation.Runner runner = DynamicSimulation.find();
 
-        Path dydFile = context.getFileSystem().getPath(line.getOptionValue(DYNAMIC_MODEL_FILE));
-        DynamicModelsSupplier dydSupplier = createMappingSupplier(dydFile, runner.getName());
+        Path dydFile = context.getFileSystem().getPath(line.getOptionValue(DYNAMIC_MODELS_FILE));
+        DynamicModelsSupplier dynamicModelsSupplier = createMappingSupplier(dydFile, runner.getName());
```
Comment (agnesLeroy) [inline dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java:147]: Hum, I'm also thinking createMappingSupplier should be renamed into createDynamicModelsSupplier to be consistent

Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java]: Done. Method createEventSupplier renamed to createEventModelsSupplier too.

File: dynamic-simulation/dynamic-simulation-dsl/src/test/groovy/com/powsybl/dynamicsimulation/groovy/DummyDynamicEventModelGroovyExtension.groovy (hunk 14 above)
Comment (marcosmc) [reply dynamic-simulation/dynamic-simulation-dsl/src/test/groovy/com/powsybl/dynamicsimulation/groovy/DummyDynamicEventModelGroovyExtension.groovy]: Done

**Pull request merged**